tests/cases/compiler/noParameterReassignmentJSIIFE.ts

```diff
@@ -0,0 +1,19 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+// @filename: index.js
+"use strict";
+
+self.importScripts = (function (importScripts) {
+  return function () {

```

Comment:
When is the synthetic argument created? Is it when a JS function references `arguments`? Are there any other conditions?

Reply:
It pulls the type from a jsdoc parameter whose name begins with `...` is present, but otherwise, yeah, that looks to be the case. It's specified in `maybeAddJsSyntheticRestParameter` (which is itself called in `getSignatureFromDeclaration`).

I think I may want to change this to only defer the type if the annotation type isn't available, since generally an annotation should override a contextual type. Problem is then I have to figure out a different fix for the annotated-but-still-technically-declarationless variant.

Reply:
Updated to handle annotations separately; not too different, we just expressly check for deferred symbols when doing contextual fixing and only sometimes defer the symbol.